Description: Use locale to figure out which week is week 1.
Author: Michael Meskes <meskes@debian.org>

diff -ru /home/michael/technik/sources/archive/bsdmainutils/usr.bin/ncal/calendar.c bsdmainutils/usr.bin/ncal/calendar.c
--- /home/michael/technik/sources/archive/bsdmainutils/usr.bin/ncal/calendar.c	2009-11-05 13:00:39.000000000 +0100
+++ bsdmainutils/usr.bin/ncal/calendar.c	2009-11-05 13:01:10.000000000 +0100
@@ -25,6 +25,7 @@
  */
 
 #include <sys/cdefs.h>
+#include <langinfo.h>
 
 #include "calendar.h"
 
@@ -261,7 +262,9 @@
 	 * otherwise this week is week 1 and the last monday is the
 	 * answer.
 	 */
-	if ((wd = weekday(nd)) > 3)
+
+	/* 3 may or may not be correct, better use what the locale says */
+	if ((wd = weekday(nd)) >= *nl_langinfo(_NL_TIME_WEEK_1STWEEK))
 		return (nd - wd + 7);
 	else
 		return (nd - wd);
diff -ru /home/michael/technik/sources/archive/bsdmainutils/usr.bin/ncal/ncal.c bsdmainutils/usr.bin/ncal/ncal.c
--- /home/michael/technik/sources/archive/bsdmainutils/usr.bin/ncal/ncal.c	2009-11-05 13:00:39.000000000 +0100
+++ bsdmainutils/usr.bin/ncal/ncal.c	2009-11-05 12:13:39.000000000 +0100
@@ -444,7 +444,13 @@
 	char    buf[80];
 
 #ifndef D_MD_ORDER
-        static int d_first = 1; /* XXX */
+        static int d_first = -1; /* XXX */
+	char * str = nl_langinfo(D_FMT);
+
+	if (d_first < 0) {
+		for (; *str && *str != 'd' && *str != 'm'; str++);
+		d_first = (*str == 'm') ? 0 : 1;
+	}
 #else
         static int d_first = -1;
 
